tests/multi_espnow/80_uasyncio_client.py

```diff
@@ -0,0 +1,125 @@
+# Test of a ESPnow echo server and client transferring data.
+# Test the ESP32 extemnsions. Assumes instance1 is an ESP32.
+# Instance1 may be and ESP32 or ESP8266
+
+try:
+    import network
+    import random
+    import usys
+    import espnow
+except ImportError:
+    print("SKIP")
+    raise SystemExit
+
+# Set read timeout to 5 seconds
+timeout = 5000
+default_pmk = b"Micropyth0nRules"
+sync = True
+
+
+def echo_server(e):
+    peers = []
+    while True:
+        peer, msg = e.irecv(timeout)
+        if peer is None:
+            return
+        if peer not in peers:
+            peers.append(peer)
+            e.add_peer(peer)
+
+        #  Echo the MAC and message back to the sender
+        if not e.send(peer, msg, sync):
+            print("ERROR: send() failed to", peer)
+            return
+
+        if msg == b"!done":
+            return
+
+
+def client_send(e, peer, msg, sync):
+    print("TEST: send/recv(msglen=", len(msg), ",sync=", sync, "): ", end="", sep="")
+    try:
+        if not e.send(peer, msg, sync):
+            print("ERROR: Send failed.")
+            return
+    except OSError as exc:
+        # Don't print exc as it is differs for esp32 and esp8266
+        print("ERROR: OSError:")
+        return
+    print("OK")
+
+
+def init(e, sta_active=True, ap_active=False):
+    wlans = [network.WLAN(i) for i in [network.STA_IF, network.AP_IF]]
+    e.active(True)
+    e.set_pmk(default_pmk)
+    wlans[0].active(sta_active)
+    wlans[1].active(ap_active)
+    wlans[0].disconnect()  # Force esp8266 STA interface to disconnect from AP
+    return e
+
+
+def poll(e):
+    poll = uselect.poll()
+    poll.register(e, uselect.POLLIN)
+    p = poll.ipoll(timeout)
+    if not p:
+        print("ERROR: poll() timeout waiting for response.")
+    return p
+
+
+# Server
+def instance0():
+    e = espnow.ESPNow()
+    init(e, True, False)
+    multitest.globals(PEERS=[network.WLAN(i).config("mac") for i in (0, 1)])
+    multitest.next()
+    print("Server Start")
+    echo_server(e)
+    print("Server Done")
+    e.active(False)
+
+
+def instance1():
+    print("SKIP")
```

Comment:
The way this skips is quite unconventional.  A simple way to do it is just require both instances to have `uasyncio` and `aioespnow` and import them at the top with the other imports.  Or do something like this:
```py
def instance1():
    try:
        import uasyncio as asyncio
        from aioespnow import AIOESPNow
    except ImportError:
        print("SKIP")
        raise SystemExit
    asyncio.run(client(AIOESPNow()))
```

Reply:
Accepted.